adam_core/propagator/pyoorb.py

```diff
@@ -282,7 +294,117 @@ def _propagate_orbits(self, orbits: Orbits, times: Time) -> Orbits:
         )
         return propagated_orbits
 
-    def _generate_ephemeris(self, orbits: Orbits, observers):
-        raise NotImplementedError(
-            "Ephemeris generation is not yet implemented for PYOORB."
+    def _generate_ephemeris(self, orbits: Orbits, observers: Observers) -> Ephemeris:
+        """
+        Generate ephemerides for orbits as viewed from observers using PYOORB.
+
+        Parameters
+        ----------
+        orbits : `~adam_core.orbits.orbits.Orbits` (N)
+            Orbits to generate ephemerides for.
+        observers : `~adam_core.observers.observers.Observers` (M)
+            Observers to generate ephemerides for.
+
+        Returns
+        -------
+        ephemeris : `~adam_core.ephemeris.ephemeris.Ephemeris` (N * M)
+            Ephemerides for each orbit as viewed from each observer.
+        """
+        # Convert orbits into PYOORB format
+        orbits_pyoorb = self._configure_orbits(
+            orbits.coordinates.values,
+            orbits.coordinates.time.to_astropy().tt.mjd,
+            OpenOrbOrbitType.CARTESIAN,
+            OpenOrbTimescale.TT,
+            magnitude=None,
+            slope=None,
         )
+
+        ephemeris_list = []
+        # Iterate over unique observatory codes and their times
+        for code_i, observer_i in observers.iterate_codes():
+            # Extract obervation times
+            times = observer_i.coordinates.time.to_astropy()
+
+            # Convert epochs into PYOORB format
+            epochs_pyoorb = self._configure_epochs(times.tt.mjd, OpenOrbTimescale.TT)
+
+            # Generate ephemeris
+            ephemeris, err = oo.pyoorb.oorb_ephemeris_full(
+                in_orbits=orbits_pyoorb,
+                in_obscode=code_i,
+                in_date_ephems=epochs_pyoorb,
+                in_dynmodel=self.dynamical_model,
+            )
+            if err != 0:
+                warnings.warn(
+                    f"PYOORB has returned error code {err} while generating ephemerides!",
+                    UserWarning,
+                )
```

Comment:
Why just warn, and not raise? Is `ephemeris` likely to be a valid value so we're able to proceed anyway?

Reply:
Yes, agreed. Better be conservative and let it fail then let something inaccurate pass through.

Reply:
These are RuntimeErrors now.